src/inference/mh.jl

```diff
@@ -146,7 +146,7 @@ function assume(spl::Sampler{<:MH}, dist::Distribution, vn::VarName, vi::VarInfo
                 spl.state.proposal_ratio -= log(cdf(stdG, (ub-r)/σ) - cdf(stdG,(lb-r)/σ))
             else # Other than Gaussian proposal
                 r = rand(proposal)
-                if (r < support(dist).lb) | (r > support(dist).ub) # check if value lies in support
+                if !(insupport(dist, r)) # check if value lies in support
```

Comment:
Thanks for the fix. Looks correct to me. Can we introduce a test for this fix? 

Maybe we can extract the code within the if-statmeent `if typeof(proposal) == Distributions.Normal{Float64}` into a function and perform the "if-else" via multiple dispatch. And by doing we can implement some tests easily too.

Reply:
It's not quite clear what we want to test :thinking: Sort, what is your `@test the_function(args...) == ?`

Reply:
It's probably ok to do the refactoring @xukai92  suggested in a separate PR